code/modules/vending/_vending.dm

```diff
@@ -858,7 +858,7 @@ GLOBAL_LIST_EMPTY(vending_products)
 /obj/machinery/vending/proc/throw_item()
 	var/obj/throw_item = null
 	var/mob/living/target = locate() in view(7,src)
-	if(!target)
+	if(!target || iseminence(target))
```

Comment:
Instead of specifically checking for eminence, you should check if the mob has a jaunt movement type.

Reply:
you mean either INCORPOREAL_MOVE_JAUNT or INCORPOREAL_MOVE_EMINENCE?

Reply:
why not ignore all INCORPOREAL including INCORPOREAL_MOVE_BASIC?

Reply:
Yes, ignore all incorporeal movement